### ISSUE TITLE
Add gradio file for local testing.

### ISSUE THREAD

**Issue by alpharesearch** [ysharma3501/LuxTTS#11]
Here is an example Gemini generated that could be added to the project. I also think having a seed for testing is really important, because as one can hear, it looks like it does make a difference. This helps to evaluate on the hardware the user has, for example on my PC with cuda I have 130x real time, but with CPU I have 5x real time. Also comparing cuda to cpu generation in some test I found cuda produces better sounding results with nothing else changed.

``` bash
pip install gradio
```

``` app.py
import gradio as gr
import soundfile as sf
import torch
import os
import numpy as np
import random
import warnings
import time
import gc # Garbage collection for memory management

# --- 1. SILENCE WARNINGS ---
warnings.filterwarnings("ignore", message=".*torch.cuda.amp.autocast.*")

# Global variables to track state
lux_tts = None
active_device = None

# --- HELPER: MODEL LOADER ---
def load_model(target_device):
    global lux_tts, active_device
    
    # Check if we actually need to reload
    if lux_tts is not None and active_device == target_device:
        return lux_tts

    # Safety check for CUDA
    if target_device == "cuda" and not torch.cuda.is_available():
        raise gr.Error("❌ CUDA (GPU) is not available on this system. Please select CPU.")

    print(f"\n🔄 Loading LuxTTS Model on [{target_device.upper()}]...")
    
    # CLEANUP: Free up memory from the old model before loading the new one
    if lux_tts is not None:
        del lux_tts
        gc.collect()
        if torch.cuda.is_available():
            torch.cuda.empty_cache()

    try:
        from zipvoice.luxvoice import LuxTTS
        # Initialize the model on the requested device
        lux_tts = LuxTTS('YatharthS/LuxTTS', device=target_device)
        active_device = target_device
        print(f"✅ Model successfully loaded on {target_device}")
        return lux_tts

    except Exception as e:
        print(f"Initialization Error: {e}")
        raise gr.Error(f"Failed to load model on {target_device}. See terminal for details.")

# --- HELPER: SEEDING ---
def set_all_seeds(seed):
    seed = int(seed)
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    if torch.cuda.is_available():
        torch.cuda.manual_seed_all(seed)
    return seed

# --- GENERATION FUNCTION ---
def clone_voice(text, ref_audio_path, rms, t_shift, num_steps, speed, return_smooth, ref_duration, seed, device_choice):
    
    if not ref_audio_path:
        raise gr.Error("Please upload a reference audio file.")

    # 1. Load/Switch Model (Only happens if device changed)
    model = load_model(device_choice)
    
    # 2. Handle Seed
    if seed == -1:
        seed = random.randint(0, 2**32 - 1)
    actual_seed = set_all_seeds(seed)
    
    print(f"\n--- Generating (Device: {device_choice} | Seed: {actual_seed}) ---")

    # 3. Encode Prompt
    # (Note: Moving the prompt to the correct device is handled internally by the library usually, 
    # but we ensure the model is on the right device first)
    encoded_prompt = model.encode_prompt(
        ref_audio_path, 
        duration=ref_duration, 
        rms=rms
    )

    # 4. Generate & Time
    start_time = time.time()
    
    final_wav = model.generate_speech(
        text, 
        encoded_prompt, 
        num_steps=int(num_steps), 
        t_shift=t_shift, 
        speed=speed, 
        return_smooth=return_smooth
    )
    
    end_time = time.time()
    generation_time = end_time - start_time

    # 5. Process Output
    audio_data = final_wav.detach().cpu().numpy().squeeze()
    
    # Clip and Convert to 16-bit PCM
    audio_data = np.clip(audio_data, -1.0, 1.0)
    audio_data = (audio_data * 32767).astype(np.int16)

    # 6. Calculate Stats
    audio_duration_sec = len(audio_data) / 48000
    speedup = audio_duration_sec / generation_time if generation_time > 0 else 0
    
    stats_msg = (
        f"💻 Device:          {device_choice.upper()}\n"
        f"⏱️ Generation Time: {generation_time:.4f} sec\n"
        f"🔊 Audio Duration:  {audio_duration_sec:.2f} sec\n"
        f"🚀 Speed:           {speedup:.1f}x Real-Time"
    )
    
    print(stats_msg)
    
    return (48000, audio_data), actual_seed, stats_msg

# --- LONGER EXAMPLE TEXT ---
long_text_example = (
    "In the heart of the ancient forest, sunlight filtered through the canopy in beams of "
    "pure gold, illuminating dust motes dancing in the still air. The only sound was the "
    "gentle murmur of a nearby brook, winding its way over mossy stones that had stood "
    "unchanged for centuries. It was a place where time seemed to slow down, allowing "
    "weary travelers to forget the burdens of the world outside."
)

# --- STARTUP: DETECT DEFAULT DEVICE ---
default_device = "cuda" if torch.cuda.is_available() else "cpu"

# --- GRADIO INTERFACE ---
with gr.Blocks(title="LuxTTS Ultimate Playground") as demo:
    gr.Markdown("# 🎙️ LuxTTS Ultimate Playground")
    gr.Markdown("Test speed, quality, and hardware performance (CPU vs GPU).")
    
    with gr.Row():
        with gr.Column():
            # Device Selection (Top for visibility)
            device_radio = gr.Radio(
                choices=["cuda", "cpu"], 
                value=default_device, 
                label="Compute Device (Switching takes a few seconds)"
            )

            text_input = gr.Textbox(
                label="Text to Speak", 
                value=long_text_example,
                lines=5
            )
            ref_audio_input = gr.Audio(
                label="Reference Audio", 
                type="filepath"
            )
            
            with gr.Accordion("Advanced Parameters", open=True):
                seed_input = gr.Number(value=42, label="Seed (-1 for Random)", precision=0)
                rms_slider = gr.Slider(0.001, 0.1, value=0.01, step=0.001, label="RMS")
                speed_slider = gr.Slider(0.1, 3.0, value=0.8, step=0.1, label="Speed")
                steps_slider = gr.Slider(1, 20, value=4, step=1, label="Num Steps")
                t_shift_slider = gr.Slider(0.1, 2.0, value=0.9, step=0.1, label="T Shift")
                duration_number = gr.Number(value=5, label="Ref Duration (sec)")
                smooth_check = gr.Checkbox(label="Return Smooth", value=False)

            generate_btn = gr.Button("Generate Voice", variant="primary")

        with gr.Column():
            output_audio = gr.Audio(label="Generated Result")
            stats_output = gr.Textbox(label="Performance Metrics", lines=5, interactive=False)
            seed_output = gr.Number(label="Actual Seed Used", interactive=False)

    # Initialize model on startup (optional, makes first generation faster)
    # logic: we just print a message, the actual load happens on first click or we can force it here.
    # To keep startup fast, we let the first click handle the load.

    generate_btn.click(
        fn=clone_voice,
        inputs=[
            text_input, ref_audio_input, rms_slider, t_shift_slider, 
            steps_slider, speed_slider, smooth_check, duration_number,
            seed_input, device_radio
        ],
        outputs=[output_audio, seed_output, stats_output]
    )

if __name__ == "__main__":
    demo.launch()
```
